### PR TITLE
terraform/kubernetes-public: add k8s-keps

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Implements: https://github.com/kubernetes/k8s.io/issues/2490
- In support of: https://github.com/kubernetes/contributor-site/pull/222

Add a world-readable bucket gs://k8s-keps along with a service account
and dedicated k8s-infra-keps@kubernetes.io group with privileged access
to the bucket and its contents.

/hold
I would like eyes on this before attempting to deploy anything to make
sure we're agreed this is the right pattern / set of infrastructure for
this use case

### CODE REVIEW

**Comment by spiffxp** [comment]
/cc @sftim @shekhar-rajak
Does this satisfy your request?
/cc @ameukam 
For the terraform changes

File: infra/gcp/terraform/k8s-infra-prow-build-trusted/prow-build-trusted/resources/test-pods/test-pods-serviceaccounts.yaml
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-prow-build-trusted/prow-build-trusted/resources/test-pods/test-pods-serviceaccounts.yaml:41]: Missing workload identity binding ?

Comment (spiffxp) [reply infra/gcp/terraform/k8s-infra-prow-build-trusted/prow-build-trusted/resources/test-pods/test-pods-serviceaccounts.yaml:41]: This is a common enough pattern that I opted to refactor into a module before adding yet more copy-pasta

I'll rebase and add the new SA/binding once https://github.com/kubernetes/k8s.io/pull/2574 merges

File: groups/sig-architecture/groups.yaml
Comment (justaugustus) [inline groups/sig-architecture/groups.yaml:16]: SIG leads + KEP tool reviewers/approvers?

File: groups/sig-architecture/groups.yaml
Comment (justaugustus) [inline groups/sig-architecture/groups.yaml:4]:
```suggestion
  # SIG Architecture infra access
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
Quick suggestions

File: groups/sig-architecture/groups.yaml
Comment (spiffxp) [reply groups/sig-architecture/groups.yaml:16]: I want to know who specifically is signing up to own and maintain this tooling and infrastructure. Not just a blanket list of people

Comment (spiffxp) [reply groups/sig-architecture/groups.yaml:16]: @sftim @shekhar-rajak
Do you have suggestions for this list as originators of the request?

@justaugustus
Are you suggesting https://github.com/kubernetes/enhancements/blob/561c1d969a7dc02d1f17b352d0e2b31953a31c53/OWNERS_ALIASES#L164-L170 ? Not sure if I've got e-mails for everyone there

**Comment by ameukam** [comment]
/lgtm
/hold
Remove when ready to deploy.

File: infra/gcp/terraform/kubernetes-public/k8s-keps.tf
Comment (ameukam) [inline infra/gcp/terraform/kubernetes-public/k8s-keps.tf:79]: Out of curious, why de we have prow on-call team ?

Comment (spiffxp) [reply infra/gcp/terraform/kubernetes-public/k8s-keps.tf:79]: If it's something prow interfaces with, I want to ensure the on-call team has sufficient access to fix whatever permissions issues may arise, since I view them as the next step up the escalation path from the team that directly owns the infrastructure

**Comment by spiffxp** [comment]
Rebased to avoid merge conflict again

/cc @jeremyrickard @johnbelamaric @justaugustus @kikisdeliveryservice @LappleApple @mrbobbytables 
I have arbitrarily chosen the members of enhancements-approvers in kubernetes/enhancements/OWNERS_ALIASES as the owners of this infrastructure

**Review by justaugustus — APPROVED** [review]
/lgtm
/approve

**Comment by kikisdeliveryservice** [comment]
> Rebased to avoid merge conflict again
> 
> /cc @jeremyrickard @johnbelamaric @justaugustus @kikisdeliveryservice @LappleApple @mrbobbytables
> I have arbitrarily chosen the members of enhancements-approvers in kubernetes/enhancements/OWNERS_ALIASES as the owners of this infrastructure

Added this to meeting agenda to see who should own this infra. Will update.

**Comment by spiffxp** [comment]
@palnabarun updated group memberships to match kep-tool-reviewers alias so you're included, thanks for pushing this forward

Looking for LGTM and then I'll remove hold / deploy when I'm next around to do so

**Comment by ameukam** [comment]
Ran `terraform apply` : 

```
Plan: 2 to add, 0 to change, 0 to destroy.
module.workload_identity_service_accounts["k8s-keps"].google_service_account.serviceaccount: Creating...
module.workload_identity_service_accounts["k8s-keps"].google_service_account.serviceaccount: Creation complete after 2s [id=projects/k8s-infra-prow-build-trusted/serviceAccounts/k8s-keps@k8s-infra-prow-build-trusted.iam.gserviceaccount.com]
module.workload_identity_service_accounts["k8s-keps"].google_service_account_iam_policy.serviceaccount_iam: Creating...
module.workload_identity_service_accounts["k8s-keps"].google_service_account_iam_policy.serviceaccount_iam: Creation complete after 1s [id=projects/k8s-infra-prow-build-trusted/serviceAccounts/k8s-keps@k8s-infra-prow-build-trusted.iam.gserviceaccount.com]
╷
│ Warning: Deprecated Attribute
│ 
│   with module.prow_build_cluster.google_container_cluster.prod_cluster,
│   on ../modules/gke-cluster/main.tf line 95, in resource "google_container_cluster" "prod_cluster":
│   95: resource "google_container_cluster" "prod_cluster" {
│ 
│ Basic authentication was removed for GKE cluster versions >= 1.19.
│ 
│ (and 3 more similar warnings elsewhere)
╵
Releasing state lock. This may take a few moments...

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

**Comment by palnabarun** [comment]
Thanks @spiffxp @bartsmykla @ameukam for working on this one! We appreciate the help! ❤️

**Comment by shekhar-rajak** [comment]
Can anyone please let me on how to use this GCP bucket to upload and fetch the uploaded file ?

**Pull request merged**
